cmd/minishift/cmd/start.go

```diff
@@ -409,26 +409,75 @@ func exportContainerImages(hostVm *host.Host, version string) {
 	fmt.Println(fmt.Sprintf("-- Exporting of OpenShift images is occuring in background process with pid %d.", exportCmd.Process.Pid))
 }
 
-// calculateDiskSizeInMB converts a human specified disk size like "1000MB" or "1GB" and converts it into Megabits
-func calculateDiskSizeInMB(humanReadableDiskSize string) int {
-	diskSize, err := units.FromHumanSize(humanReadableDiskSize)
+type sizeValidatorFunc func(name string, diskSize string) error
+type sizeConverterFunc func(input int64) int
+
+func calculateHumanSize(sizeValidator sizeValidatorFunc, name string, humanReadableSize string, unit string, sizeConverter sizeConverterFunc) (int, error) {
```

Comment:
I am intrigued by the use of funcs as method parameters. What do you really gain here? IMO it just gets harder to understand the code.